tensorflow/compiler/mlir/xla/tests/buffer-assignment-positions.mlir

```diff
@@ -0,0 +1,131 @@
+}
+
+// -----
+
+// CHECK-LABEL: Testing : invCriticalEdge
+func @invCriticalEdge(%cond : i1, %arg0 : tensor<2xf32>) -> tensor<2xf32>{
+  // CHECK: Alloc: %0 = "xla_hlo.exponential"
+    %0 = "xla_hlo.exponential"(%arg0) : (tensor<2xf32>) -> tensor<2xf32>
+    cond_br %cond, ^bb1, ^exit(%arg0 : tensor<2xf32>)
+  ^bb1:
+    br ^exit(%0 : tensor<2xf32>)
+  ^exit(%arg1: tensor<2xf32>):
+    return %arg1 : tensor<2xf32>
+  // CHECK-NEXT: Dealloc: return
+}
+
+// -----
+
+// CHECK-LABEL: Testing : ifElse
+func @ifElse(%cond : i1, %arg0 : tensor<2xf32>) -> tensor<2xf32>{
+  // CHECK: Alloc: %0 = "xla_hlo.exponential"(%arg1)
+    %0 = "xla_hlo.exponential"(%arg0) : (tensor<2xf32>) -> tensor<2xf32>
+    cond_br %cond, ^bb1(%arg0, %0: tensor<2xf32>, tensor<2xf32>), ^bb2(%0, %arg0: tensor<2xf32>, tensor<2xf32>)
+  ^bb1(%arg1 : tensor<2xf32>, %arg2 : tensor<2xf32>):
+    br ^exit(%arg1, %arg2 : tensor<2xf32>, tensor<2xf32>)
+  ^bb2(%arg3 : tensor<2xf32>, %arg4 : tensor<2xf32>):
+    br ^exit(%arg3, %arg4 : tensor<2xf32>, tensor<2xf32>)
+  ^exit(%arg5 : tensor<2xf32>, %arg6 : tensor<2xf32>):
+  // CHECK-NEXT: Dealloc: %7 = "xla_hlo.exponential"(%5)
+  // CHECK: Alloc: %7 = "xla_hlo.exponential"(%5)
+  // CHECK-NEXT: Dealloc: return
+    %1 = "xla_hlo.exponential"(%arg5) : (tensor<2xf32>) -> tensor<2xf32>
+    return %1 : tensor<2xf32>
+}
+
+// -----
+
+// CHECK-LABEL: Testing : ifElseNoUsers
+func @ifElseNoUsers(%cond : i1, %arg0 : tensor<2xf32>) -> tensor<2xf32>{
+  // CHECK: Alloc: %0 = "xla_hlo.exponential"(%arg1)
+    %0 = "xla_hlo.exponential"(%arg0) : (tensor<2xf32>) -> tensor<2xf32>
+    cond_br %cond, ^bb1(%arg0, %0: tensor<2xf32>, tensor<2xf32>), ^bb2(%0, %arg0: tensor<2xf32>, tensor<2xf32>)
+  ^bb1(%arg1 : tensor<2xf32>, %arg2 : tensor<2xf32>):
+    br ^exit(%arg1, %arg2 : tensor<2xf32>, tensor<2xf32>)
+  ^bb2(%arg3 : tensor<2xf32>, %arg4 : tensor<2xf32>):
+    br ^exit(%arg3, %arg4 : tensor<2xf32>, tensor<2xf32>)
+  ^exit(%arg5 : tensor<2xf32>, %arg6 : tensor<2xf32>):
+  // CHECK-NEXT: return
+    return %arg0 : tensor<2xf32>
+}
+
+// -----
+
+// CHECK-LABEL: Testing : ifElseNested
+func @ifElseNested(%cond : i1, %arg0 : tensor<2xf32>) -> tensor<2xf32>{
+  // CHECK: Alloc: %0 = "xla_hlo.exponential"(%arg1)
+    %0 = "xla_hlo.exponential"(%arg0) : (tensor<2xf32>) -> tensor<2xf32>
+    cond_br %cond, ^bb1(%arg0, %0: tensor<2xf32>, tensor<2xf32>), ^bb2(%0, %arg0: tensor<2xf32>, tensor<2xf32>)
+  ^bb1(%arg1 : tensor<2xf32>, %arg2 : tensor<2xf32>):
+    br ^exit(%arg1, %arg2 : tensor<2xf32>, tensor<2xf32>)
+  ^bb2(%arg3 : tensor<2xf32>, %arg4 : tensor<2xf32>):
+    cond_br %cond, ^bb3(%arg3 : tensor<2xf32>), ^bb4(%arg4 : tensor<2xf32>)
+  ^bb3(%arg7 : tensor<2xf32>):
+    br ^exit(%arg7, %arg3 : tensor<2xf32>, tensor<2xf32>)
+  ^bb4(%arg8 : tensor<2xf32>):
+    br ^exit(%arg3, %arg8 : tensor<2xf32>, tensor<2xf32>)
+  ^exit(%arg5 : tensor<2xf32>, %arg6 : tensor<2xf32>):
+  // CHECK-NEXT: Dealloc: %9 = "xla_hlo.exponential"(%7)
```

Comment:
using %7, %9, %0 in CHECK is not the best thing to do. Please, try to use `{{.*}}` at least, or, better, capture with some meaningful names.

Reply:
Done.